web/Controllers/LandlordController.cs

```diff
@@ -141,4 +141,34 @@ public IActionResult ViewProperties()
         var listOfProperties = databaseResult.Select(PropertyViewModel.FromDbModel).ToList();
         return View("Properties", new PropertiesDashboardViewModel(listOfProperties));
     }
+
+    [HttpGet]
+    [Route("edit")]
+    public async Task<ActionResult> EditProfilePage(int landlordId)
+    {
+        var user = GetCurrentUser();
+        var landlordFromDb =  await _landlordService.GetLandlordIfExistsFromId(landlordId);
+        if (landlordFromDb == null) return StatusCode(404);
+        if (user.LandlordId != landlordFromDb.Id && !user.IsAdmin) return StatusCode(403);
+        return View("EditProfilePage", LandlordProfileModel.FromDbModel(landlordFromDb, user));
+    }
+
+    [HttpPost]
+    [Route("edit")]
+    public async Task<ActionResult> EditProfileUpdate([FromForm] LandlordProfileModel editModel)

```

Comment:
One more thing, should check `ModelState.IsValid` here. Fine to put in your other MR if you want to get this merged, sorry didn't think of it earlier.

Reply:
Ahhh yeah good point! I will merge and add this to the other MR